ateam_geometry/src/intersection.cpp

```diff
@@ -74,6 +75,50 @@ std::optional<std::variant<Point, std::pair<Point, Point>>> intersection(
   }
 }
 
+std::optional<std::variant<Point, std::pair<Point, Point>>> intersection(
+  const Circle & circle,
+  const Segment & segment)
+{
+
+  const auto maybe_intersection = intersection(circle, segment.supporting_line());
+
+  if (!maybe_intersection.has_value()) {
+    return std::nullopt;
+  }
+
+  const auto intersection = maybe_intersection.value();
+
+  if (std::holds_alternative<Point>(intersection)) {
+    const auto intersection_point = std::get<Point>(intersection);
+
+    if (CGAL::do_intersect(segment, intersection_point)) {
+      return intersection_point;
+    }
```

Comment:
Does this approach handle the floating point mismatches well? It feels odd to use `do_intersect` in this branch and the distance check in the branch below.